src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1545,32 +1465,35 @@ def compute_anchor_edit_info(
             anchor_info[anchor_id].add(fix)
         return dict(anchor_info)
 
-    def _validate_segment_after_fixes(self, rule_code, dialect, fixes_applied, segment):
+    def _validate_segment_after_fixes(
+        self,
+        rule_code: str,
+        dialect: "Dialect",
+        fixes_applied: List[LintFix],
+        segment: BaseSegment,
+    ) -> None:
         """Checks correctness of new segment against match or parse grammar."""
         ctx = ParseContext(dialect=dialect)
         try:
             # :HACK: Calling parse() corrupts the segment 'r'
             # in some cases, e.g. adding additional Dedent child
             # segments. Here, we work around this by calling
             # parse() on a "backup copy" of the segment.
-            r_copy = deepcopy(segment)
-            for seg in r_copy.segments:
-                seg.pos_marker = replace(
-                    seg.pos_marker,
-                    templated_file=self.pos_marker.templated_file,
-                )
-            r_copy.parse(ctx)
+            segment_copy = segment.copy()
+            segment_copy.parse(ctx)
```

Comment:
The builtin `.copy()` method is now a lot more robust, so it should work fine in place of `deepcopy()`.